packages/mockserver-config-writer/src/mockserver-config/ui5-mock-yaml.ts

```diff
@@ -0,0 +1,124 @@
+ *
+ * ui5-mock.yaml exists
+ * -----------------------------
+ * Update or create middleware configuration for 'sap-fe-mockserver' with the latest configuration.
+ * Other parts of the ui5-mock.yaml will not be modified.
+ *
+ * ui5-mock.yaml does not exist
+ * -----------------------------
+ * If ui5.yaml exists, copy it and add/update the 'sap-fe-mockserver' middleware configuration.
+ * If ui5.yaml does not exist, create a new ui5-mock.yaml file with 'sap-fe-mockserver' middleware
+ * configuration and basic default configuration for 'fiori-tools-proxy' and 'fiori-tools-appreload'.
+ *
+ * @param fs - mem-fs reference to be used for file access
+ * @param basePath - path to project root, where package.json and ui5.yaml is
+ * @param webappPath - path to webapp folder, where manifest.json is
+ * @param config - optional config passed in by consumer
+ */
+export async function enhanceYaml(
+    fs: Editor,
+    basePath: string,
+    webappPath: string,
+    config?: Ui5MockYamlConfig
+): Promise<void> {
+    const ui5MockYamlPath = join(basePath, 'ui5-mock.yaml');
+    let mockConfig;
+    const manifest: Manifest = JSON.parse(fs.read(join(webappPath, 'manifest.json')));
+    const mockserverPath = config?.path || getMainServiceDataSource(manifest)?.uri;
+
+    if (fs.exists(ui5MockYamlPath)) {
+        mockConfig = await updateUi5MockYamlConfig(fs, ui5MockYamlPath, mockserverPath);
+    } else {
+        mockConfig = fs.exists(join(basePath, 'ui5.yaml'))
+            ? await generateUi5MockYamlBasedOnUi5Yaml(fs, basePath, mockserverPath)
+            : await generateNewUi5MockYamlConfig(manifest['sap.app']?.id || '', mockserverPath);
+    }
+    const yaml = mockConfig.toString();
+    fs.write(ui5MockYamlPath, yaml);
+}
+
+/**
+ * Update existing ui5-mock.yaml config. This will add or replace existing middleware configuration
+ * 'sap-fe-mockserver' with state of the art config.
+ *
+ * @param fs - Editor instance to read existing information
+ * @param ui5MockYamlPath - path to ui5-mock.yaml file
+ * @param [path] - optional url path the mockserver listens to
+ */
+async function updateUi5MockYamlConfig(fs: Editor, ui5MockYamlPath: string, path?: string): Promise<UI5Config> {
+    const existingUi5MockYamlConfig = await UI5Config.newInstance(fs.read(ui5MockYamlPath));
+    existingUi5MockYamlConfig.updateCustomMiddleware(await getMockserverMiddleware(path));
+    return existingUi5MockYamlConfig;
+}
+
+/**
+ * Create a new ui5-mock.yaml based on existing ui5.yaml.
+ *
+ * @param fs - Editor instance to read existing information
+ * @param basePath -
+ * @param [path] - optional path for mockserver config
+ */
+async function generateUi5MockYamlBasedOnUi5Yaml(fs: Editor, basePath: string, path?: string): Promise<UI5Config> {
+    const ui5MockYamlConfig = await UI5Config.newInstance(fs.read(join(basePath, 'ui5.yaml')));
+    ui5MockYamlConfig.updateCustomMiddleware(await getMockserverMiddleware(path));
```

Comment:
What does update do if it doesn't exist? Does it add it?

Reply:
Yes, it adds it. I've added this recently  `@sap-ux/ui5-config`

https://github.com/SAP/open-ux-tools/blob/46b8a2fe8db734da1b6f713d2bcca55310a59d77/packages/ui5-config/src/ui5config.ts#L352-L360